http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala

```diff
@@ -168,6 +173,103 @@ class AzurePubsubHandlerInterp[F[_]: Parallel](
+
+    _ <- monoOpt match {
+      case None =>
+        logger.error(ctx.loggingCtx)(
+          s"Cannot find runtime ${runtime.projectNameString}."
+        ) // Error? For now, log a message
+      case Some(mono) =>
+        // Move to top of file - reason being if composed with another same named variable it could cause confusion
+        implicit val isJupyterUpDoneCheckable: DoneCheckable[Boolean] = (v: Boolean) => v
+        val task = for {
+          // TODO: Add timeout. See how long it takes to start.
+          // 10 minutes tops. If startup script it may take longer
+          _ <- F.blocking(mono.block())
+          // Once mono is complete means vm is running.
+          // Update DB once completed.
+          // Even when Mono completes, we can't assume it's running
+          // Need to check if JupyterLab is running
+          isJupyterUp = jupyterDAO.isProxyAvailable(runtime.cloudContext, runtime.runtimeName)
+          _ <- streamUntilDoneOrTimeout(
+            isJupyterUp,
+            config.createVmPollConfig.maxAttempts,
+            config.createVmPollConfig.interval,
+            s"Jupyter was not running within ${config.createVmPollConfig.maxAttempts} attempts with ${config.createVmPollConfig.interval} delay"
+          )
+          _ <- clusterQuery.updateClusterStatus(runtime.id, RuntimeStatus.Running, ctx.now).transaction
+          _ <- logger.info(ctx.loggingCtx)("runtime is ready")
+        } yield ()
+        asyncTasks.offer(
+          Task(
+            ctx.traceId,
+            task,
+            // Set status to error? Probably not, will get user stuck.
+            // Front leo will set to starting. If we don't change anything, status will stay starting.
+            Some(e =>
+              handleAzureRuntimeStartError(
+                AzureRuntimeStartingError(runtime.id, s"starting runtime ${runtime.projectNameString} failed", e),
+                ctx.now
+              )
+            ),
+            ctx.now,
+            "startRuntime"
+          )
+        )
+    }
+  } yield ()
+
+  // TODO: Monitor?
+  // TODO: Flush welder cache?
+  override def stopAndMonitorRuntime(runtime: Runtime, azureCloudContext: AzureCloudContext)(implicit
+    ev: Ask[F, AppContext]
+  ): F[Unit] = for {
+    ctx <- ev.ask
+    monoOpt <- azureVmServiceInterp.stopAzureVm(InstanceName(runtime.runtimeName.asString), azureCloudContext)
+    _ <- monoOpt match {
+      case None =>
+        logger.error(ctx.loggingCtx)(
+          s"Cannot find runtime ${runtime.projectNameString}."
+        ) // Error? For now, log a message
+      case Some(mono) =>
+        val task = for {
+          _ <- F.blocking(mono.block()) // TODO: Add timeout
+          _ <- clusterQuery.updateClusterStatus(runtime.id, RuntimeStatus.Stopped, ctx.now).transaction
+          // TODO: Do we NONE the IP address like GCP?
+          // _ <- clusterQuery.updateClusterHostIp(params.runtime.id, None, ctx.now).transaction
```

Comment:
GCP logic performs this query. Should we do this for Azure as well?

Reply:
we don't need to...the reason why we need to do this for GCE is becuz everytime a VM starts/stops, the host IP will change....but for Azure, we're accessing it via a relay URL, which won't change during start/stop cycle